src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Razor/IHelperMatcher.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Razor
+{
+    public interface IHelperMatcher
+    {
+        Task<IEnumerable<HelperReplacement>> GetHelperReplacementsAsync(RazorCodeDocument document);
```

Comment:
IAsyncEnumerable?

Reply:
Still not sure about when to use one over the other, but I like the idea of lazy calculations when possible

Reply:
Looking at usage, you could get rid of the `HasHelperAsync` with an IAsyncEnumerable and just do an `AnyAsync` on it

Reply:
Yea, that's a better API shape. I'll make that update.

Reply:
I've updated IHelperMatcher to have a single method returning an enumerable.